addons/crm/wizard/crm_lead_to_opportunity.py

```diff
@@ -82,6 +79,12 @@ def _compute_duplicated_lead_ids(self):
                 convert.lead_id.partner_id.email if convert.lead_id.partner_id.email else convert.lead_id.email_from,
                 include_lost=True).ids
 
+    @api.depends('partner_id')
+    def _compute_company_partner_id(self):
+        for convert in self.filtered('partner_id'):
+            if convert.company_partner_id and convert.partner_id.parent_id and not convert.company_partner_id.filtered_domain([('id', 'parent_of', convert.partner_id)]):
+                convert.company_partner_id = convert.partner_id.parent_id
```

Comment:
That's strange, are people going to be able to break hierarchy by changing it ? It should probably not be displayed if the contact has a parent, don't let people break everything in db.